### ISSUE TITLE
Can't connect over SSL/TLS after upgrade

### ISSUE THREAD

**Issue by beard7** [Adldap2/Adldap2-Laravel#814]
- Laravel Version: 5.8
- Adldap2-Laravel Version: 6.0
- PHP Version: 7.3
- LDAP Type: ActiveDirectory

### Description:
After upgrading from Adldap2 5.0 to 6.0, I can no longer connect to our AD controllers securely.

I've tried using all combinations of ports 636 and 3269 and TLS and SSL, but I always get the error:
`LDAP (ldaps://x.x.x.x:3269 ldaps://x.x.x.x:3269) - Connection: default - Operation: Failed - Username: DOM\user - Reason: Can't contact LDAP server`

If I disable SSL/TLS and use port 389 everything is working.

Also, with version 5.0 I had to use the following in my `config/ldap.php` file:
    'custom_options' => [
        LDAP_OPT_X_TLS_REQUIRE_CERT => LDAP_OPT_X_TLS_NEVER
     ]
but this is not solving the problem with version 6.0

Any suggestions? Thanks

**Comment by Sebi94nbg** [comment]
First of all, you need to figure out, what's the reason for this issue.

Most of the time, it's due to a SSL certification issue. I always create a simple clean and plain PHP script and execute this from a command line. This shows you the exact reason for the failed SSL connection.

I always add this line to my `/etc/ldap/ldap.conf` in order to avoid "failed to verify SSL certificate" issues: `TLS_REQCERT never`

Explanation of this setting:
> Client will not request a server certificate, but if received will ignore and continue connection.

Related to #813.

**Comment by stevebauman** [comment]
I believe this was actually a bug with Adldap2-Laravel v5.0 that used Adldap2 v9.0.

This was fixed in the [v10.0.11](https://github.com/Adldap2/Adldap2/releases/tag/v10.0.11) release:

> ldap_start_tls() failures are now converted to ConnectionException - d376888

In v9.0, TLS errors failed silently due to the muting of the `bind()` call in the `Adldap\Auth\Guard`, as some users PHP error warning levels differed and they were not being converted to exceptions - and as such, did not break execution of the `bind` call. The connection would continue to work due to port 389 being the standard port for non-TLS connections.

As mentioned above by @Sebi94nbg, if you're running a Windows server, you will have to create the following directory on the `C:\` disk on your server and add a file named `ldap.conf` with the contents:

**Path:** `C:\openldap\sysconf\ldap.conf`

**Contents:**
````
TLS_REQCERT never
```

That should resolve your issue. Let me know if you encounter anything further. :smile:
